src/SmoFacade/Database.cs

```diff
@@ -76,13 +76,15 @@ public List<BackupMetadata> RecentBackups()
                   "FROM msdb.dbo.backupset s " +
                   "INNER JOIN msdb.dbo.backupmediafamily m ON s.media_set_id = m.media_set_id " +
                   "WHERE s.last_lsn >= (" +
-                  "SELECT MAX(last_lsn) FROM msdb.dbo.backupset " +
-                  "WHERE [type] = 'D' " +
-                  "AND database_name = @dbName " +
-                  "AND is_copy_only = 0" +
+                    "SELECT MAX(last_lsn) FROM msdb.dbo.backupset " +
+                    "WHERE [type] = 'D' " +
+                    "AND database_name = @dbName " +
+                    "AND is_copy_only = 0" +
+                    "AND server_name = @@ServerName " +
                   ") " +
                   "AND s.database_name = @dbName " +
-                  "AND is_copy_only = 0" +
+                  "AND is_copy_only = 0 " +
+                  "AND s.server_name = @@ServerName " +

```

Comment:
Can you put it in all caps , that's how the t-sql is defined.
https://docs.microsoft.com/en-us/sql/t-sql/functions/servername-transact-sql?view=sql-server-ver15

Reply:
Sure, done!